packages/react-components/src/Concepts/Positioning/PositioningPopperImperativeHandle.stories.tsx

```diff
@@ -6,13 +6,16 @@ import { PositioningProps } from '@fluentui/react-positioning';
 export const PopperImperativeHandle = () => {
   const [loading, setLoading] = React.useState(true);
   const popperRef: PositioningProps['popperRef'] = React.useRef({ updatePosition: () => null });
-  const timeoutRef = React.useRef(0);
+  const timeoutRef = React.useRef<ReturnType<typeof setTimeout>>();
```

Comment:
@Hotell Now Node typings are leaking to stories, not sure that it's okay 😯

Reply:
so because the custom logic that is being provided in root `/.storybook` and in `react-components` - (main.utils.js), which uses `node` APIs, node typings are being included to the checking program ( which is how TS works ). At this point there is no functionality that allows us to prevent that.

Only solution is to be explicit and stop doing antipatterns like accesing globals directly.

**Instead:**

```ts
timeoutRef.current = setTimeout(() => setLoading(false), 1000);
```

**Do:**

```ts
timeoutRef.current = window.setTimeout(() => setLoading(false), 1000);
```

**This will resolve the type issues you're having.**

---

Another thingy:

Now because I introduced tests in `react-components/.storybook` , we would need to add `jest` glboals to `react-components/.storybook/tsconfig.json`.  That will pollute all stories, thus allowing anyone to use jest globals in stories. I don't think that's the proper solution ATM, so instead we can do following for that test file:

```diff
@filename react-components/.storybook/main.utils.test.js

+/// <reference types="jest" />
const utils = require('./main.utils');

describe(`main utils`, () => {

```

Reply:
Is using `<reference types="jest" />` is a practice that we should follow in such cases?

Reply:
ideal scenario is to have this in separate package (logic,tests) with proper globals set.  but as a temporary workaround/corner case I think so, yes.

Why? 
- tests are not shipped not exported so not used in some other context, thus it is ok to specify local globals inline, if not specified on package level

Reply:
also going forward ( once we migrate to jest 27 ) we should stop using globals as jest provides imports for their API ( one less global in our lives yay )